backend/reservations/views/rooms.py

```diff
@@ -266,22 +306,44 @@ def swap_it_up(request):
             logger.warning("[-] No room matching code")
             return Response("No room matching that code", status=status.HTTP_400_BAD_REQUEST)
 
+        if swap_room_mine.name_take3 != swap_room_theirs.name_take3:
+            logger.warning("Attempt to swap mismatched room types %s (%s) - %s (%s)",
+                           swap_room_mine.number, swap_room_mine.name_take3,
+                           swap_room_theirs.number, swap_room_theirs.name_take3)
+            return Response("Unable to swap rooms", status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Add info about why unable to swap, eg
`"Unable to swap rooms: not matching room type"`

Reply:
This is the sort of information we shouldn't bubble up to end users because it gives them information they could use to circumvent these checks. We _do_ log the details as a warning though.

Reply:
How could they use it? It should be clear that they shouldnt be trying to swap different types of rooms.

Reply:
The only way that a user should ever see this error message is if they are using the API directly. And if they are doing that, we should not make it easy for them.